File: Java/Searching-Algorithms/LinearSearch.java
package LinearSearchAlgorithm;

public class Q1Searching {
    public static void main(String[] args) {

        // in Linear Search :
        // In the best case O(1) : the element will be found be
        // found  on the first try when 1st element == target {i.e,
        // It will make only 1 comparison }
        // In the worst case O(N) : It will make n
        // comparison only  where n is the size of the array



        // search in the array :return the index if item found
        // otherwise if the item not found return -1
        int[] arr ={22,44,97,67,92,17};
        int ans = linearSearch(arr,97);
        System.out.println(ans);
    }

    static int linearSearch(int[] arr, int target) {
        if(arr.length ==0){
            return -1;
        }

        // run a for loop
        for(int index=0;index<arr.length;index++) {
          // check for element at every index if it is = target
            if(arr[index] == target){
                return index;
            }
        }
        // this line will execute if none of the return statements above have executed
        // hence the target not found
        return -1;
    }
}

File: Python/Searching-Algorithms/linear-search.py
#Linear Search Program
def linear_search(l,key):
  for i in l:
    if i==key:
      return i
  return -1
l=[1,5,2,7,3]
k=2
f=linear_search(l,k)
if f!=-1:
  print("Key element is at %d index"%f)
else:
  print("Element not found")
#if k=(any number which is not in the list)
#Output: Element not found
#Otherwise it prints "Key element is at {index of that element in list} index"
